### ISSUE TITLE
Internal query batching fails when the table name is 'stores'

### ISSUE THREAD

**Issue by roshan1988** [prisma/prisma#20324]
### Bug description

I have one of my table named 'stores' in postgres.

When there is a **findUnique** for stores, the query works fine, and returns the value correctly.

But when there are **multiple** **findUniques** for the 'stores' table, and when prisma is internally batching them, client is returning null value for all of the calls.

This happens only when I move to the jsonProtocol supported versions of prisma.

The internal queries seems to be created correctly by prisma though.

`
SELECT "public"."stores"."id", "public"."stores"."displayName", "public"."stores"."subDomain", "public"."stores"."widgets", "public"."stores"."ownerId", "public"."stores"."createdBy", "public"."stores"."status", "public"."stores"."createdAt", "public"."stores"."updatedAt", "public"."stores"."fbPixelId", "public"."stores"."brandId", "public"."stores"."description", "public"."stores"."profileImageId", "public"."stores"."isCommerceEnabled", "public"."stores"."isVerified", "public"."stores"."isOutOfStock", "public"."stores"."type", "public"."stores"."appVisibility" FROM "public"."stores" WHERE (("public"."stores"."id" = $1 AND "public"."stores"."status" = $2) OR ("public"."stores"."id" = $3 AND "public"."stores"."status" = $4)) OFFSET $5
`

So I am guessing something is going wrong while mapping them back to the prisma models and returning null

### Prisma information

```prisma
model Stores {
  id                     String                   @id @default(cuid())
  displayName            String
  subDomain              String                   @unique
  widgets                Json?
  ownerId                String
  createdBy              String
  status                 StoreStatus              @default(ACTIVE)
  createdAt              DateTime                 @default(now())
  updatedAt              DateTime                 @updatedAt
  fbPixelId              String?
  brandId                String?
  description            String?
  profileImageId         String?
  isCommerceEnabled      Boolean                  @default(true)
  isVerified             Boolean                  @default(false)
  isOutOfStock           Boolean                  @default(false)
  type                   StoreType                @default(CREATOR)
  appVisibility          StoreAppVisibility       @default(OPEN)
  Brand                  Brands?                  @relation(fields: [brandId], references: [id])
  CreatedBy              Users                    @relation("StoreCreatedBy", fields: [createdBy], references: [id])
  Owner                  Users                    @relation("StoreOwner", fields: [ownerId], references: [id])
  ProfileImage           Images?                  @relation(fields: [profileImageId], references: [id])
  ArchivedStore          ArchivedStores?
  CreatorBarters         Barters[]                @relation("BarterBrandStore")
  BrandBarters           Barters[]                @relation("BarterCreatorStore")
  AttributeCartItems     CartItems[]
  Followings             Followings[]
  OrderCouponExpenses    OrderCouponExpenses[]
  OrderItemEarnings      OrderItemEarnings[]
  AttributeOrderItems    OrderItems[]
  SharedStoreCollections SharedStoreCollections[]
  SocialPosts            SocialPosts[]
  StoreCollections       StoreCollections[]
  StoreInstagram         StoreInstagrams?
  StoreProducts          StoreProducts[]
  Products               Products[]
  StoreSocialProfiles    StoreSocialProfiles[]
  StoreYoutube           StoreYoutubes?
  UserStoreRoles         UserStoreRoles[]
  WishListItems          WishlistItems[]
  OrderReports           StoreOrderReports[]
  PostComments           PostComments[]
  ProductReviews         ProductReviews[]
  CreatorRewards         CreatorRewards[]
  StoreBlockings         StoreBlockings[]
  StoreCoupons           Coupons[]

  @@index([ownerId, status])
  @@map("stores")
}
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: All envioronments
- Database: PostgreSQL
- Node.js version: 18.5

### Prisma Version

```
5.0.0
```

**Comment by millsp** [comment]
Thanks for the report @roshan1988. Could you please also share the `findUnique` query that you are using? I was not able to simply reproduce in a simple project, so the reproduction may be more involved (I also tried with a different schema). Could you please provide us with a small repro (with a part of your schema, and the code triggering this)?

```ts
const data = await Promise.all([
    prisma.user.findUnique({ where: { id: 1 } }),
    prisma.user.findUnique({ where: { id: 2 } }),
    prisma.user.findUnique({ where: { id: 3 } }),
    prisma.user.findUnique({ where: { id: 4 } }),
    prisma.user.findUnique({ where: { id: 5 } }),
    prisma.user.findUnique({ where: { id: 6 } }),
    prisma.user.findUnique({ where: { id: 7 } }),
    prisma.user.findUnique({ where: { id: 8 } }),
    prisma.user.findUnique({ where: { id: 9 } }),
])
```

**Comment by SevInf** [comment]
Hi @roshan1988.
It looks to me like the name of the model has nothing to do with it, and this is issue is rather a duplicate of #20227. Could you check if still happens on `5.1.0-dev.15` snapshot version, mentioned in the issue? We don't recommend using dev versions in production, but this will help to ensure that you problem is fixed in Prisma 5.1

**Comment by roshan1988** [comment]
> Hi @roshan1988. It looks to me like the name of the model has nothing to do with it, and this is issue is rather a duplicate of #20227. Could you check if still happens on `5.1.0-dev.15` snapshot version, mentioned in the issue? We don't recommend using dev versions in production, but this will help to ensure that you problem is fixed in Prisma 5.1

Thanks @SevInf for the reply. I am able to make this work with 5.1.0-dev.15. Will wait for the official 5.1 release

**Comment by SevInf** [comment]
Thanks for getting back to us and confirming it is fixed!
5.1 will be released in a week and since the problem is solved, I am going to close this issue now.
